### PR TITLE
Fix changes on master-da0bbfa

### PR DESCRIPTION
**Pull request by canonbrother**
#### What kind of PR is this?:

/kind fix
/kind chore

### CODE REVIEW

File: packages/jellyfish-api-core/__tests__/category/loan/listVaults.test.ts [hunk 1]
```diff
@@ -103,26 +104,32 @@ describe('Loan listVaults', () => {
     await testing.generate(1)
 
     // List vaults
-    // should be only vaultId1, vaultId2, vaultId3 coz isUnderLiquidation filter default value equals to false
+    // should be only vaultId1, vaultId2, vaultId3, vaultId4 coz isUnderLiquidation filter default value equals to false
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/listVaults.test.ts:107]: looks like this is not true with lastest cpp side developments. default state is `VaultState::Unknown`

File: packages/jellyfish-api-core/__tests__/category/loan/listVaults.test.ts [hunk 2]
```diff
@@ -223,112 +238,112 @@ describe('Loan listVaults with options and pagination', () => {
...
   it('should listVaults with pagination', async () => {
     // List vaults
     const vaults = await testing.rpc.loan.listVaults({ limit: 2 })
-    expect(Object.keys(vaults).length).toStrictEqual(2)
+    expect(vaults.length).toStrictEqual(2)
 
     // fetch the second page
-    const vaultsSecondPage = await testing.rpc.loan.listVaults({ including_start: false, start: vaults[Object.keys(vaults).length - 1].vaultId })
+    const vaultsSecondPage = await testing.rpc.loan.listVaults({ including_start: false, start: vaults[vaults.length - 1].vaultId })
     // should be 1 entries
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/listVaults.test.ts:315]:
```suggestion
    // should be 2 entries
```

File: packages/jellyfish-api-core/__tests__/category/loan/listVaults.test.ts [hunk 3]
```diff
@@ -223,112 +238,112 @@ describe('Loan listVaults with options and pagination', () => {
...
-    expect(Object.keys(vaultsSecondPage).length).toStrictEqual(1)
+    expect(vaultsSecondPage.length).toStrictEqual(2) // total 4, started at index[2], listing 2
 
     // fetch the second page with including_start = true
-    const vaultsSecondPageIncludingStart = await testing.rpc.loan.listVaults({ including_start: true, start: vaults[Object.keys(vaults).length - 1].vaultId })
+    const vaultsSecondPageIncludingStart = await testing.rpc.loan.listVaults({ including_start: true, start: vaults[vaults.length - 1].vaultId })
     // should be 2 entries
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/loan/listVaults.test.ts:320]:
```suggestion
    // should be 3 entries
```

File: packages/jellyfish-api-core/src/category/loan.ts [hunk 4]
```diff
@@ -240,11 +240,15 @@ export class Loan {
    * @param {ListVaultOptions} [options]
    * @param {string} [options.ownerAddress] Address of the vault owner
    * @param {string} [options.loanSchemeId] Vault's loan scheme id
-   * @param {boolean} [options.isUnderLiquidation = false] vaults under liquidation
-   * @return {Promise<ListVaultDetails[]>} Array of objects including details of the vaults.
+   * @param {boolean} [options.state = 'Collateralized'] vault's state
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/loan.ts:243]:
```suggestion
    * @param {VaultState} [options.state = VaultState.UNKNOWN] vault's state
```

**Review by surangap — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Comment by fuxingloh** [comment]
Since list vault and get vault return different data interface, we should separate out the iteratces

**Pull request merged**